### PR TITLE
DL-199: Be able to support filesystem-path like name

### PR DESCRIPTION
**Pull request by sijie**
In order to support hierarchical namespace, we need to be able to support filesystem path like log name.

### CODE REVIEW

File: distributedlog-core/src/main/java/org/apache/distributedlog/util/DLUtils.java [hunk 1]
```diff
@@ -281,41 +281,89 @@ public static boolean isReservedStreamName(String name) {
...
-                reason = "invalid charater @" + i;
-                break;
-            }
+        if (isReservedStreamName(logName)) {
+            throw new InvalidStreamNameException(logName, "Log Name is reserved");
         }
-        if (null != reason) {
-            throw new InvalidStreamNameException(nameOfStream, reason);
+
+        if (logName.charAt(0) == 47) {
```
Comment (jiazhai) [inline distributedlog-core/src/main/java/org/apache/distributedlog/util/DLUtils.java]: Is 47 intentional?  How about replace 47 with '/'  in this file?

File: distributedlog-core/src/main/java/org/apache/distributedlog/util/DLUtils.java [hunk 2]
```diff
@@ -281,41 +281,89 @@ public static boolean isReservedStreamName(String name) {
...
+                        reason = "null character not allowed @" + i;
+                        break;
+                    }
+
+                    if (c == '<' || c == '>') {
+                        reason = "< or > specified @" + i;
+                        break;
+                    }
+
+                    if (c == ' ') {
```
Comment (jiazhai) [inline distributedlog-core/src/main/java/org/apache/distributedlog/util/DLUtils.java]: Is it needed other empty char like  HT, VT, CR, NL('\t',  '\r', '\n', '\x0b') ?

File: distributedlog-core/src/test/java/org/apache/distributedlog/TestBKDistributedLogNamespace.java [hunk 3]
```diff
@@ -143,7 +172,7 @@ public void testInvalidStreamName() throws Exception {
         }
 
         try {
-            namespace.openLog("/test2");
+            namespace.openLog("/ test2");
             fail("should fail to create invalid stream /test2");
```
Comment (jiazhai) [inline distributedlog-core/src/test/java/org/apache/distributedlog/TestBKDistributedLogNamespace.java]: seems also need a space here: / test2

**Comment by sijie** [comment]
@leighst @yzang : mind reviewing this one?

File: distributedlog-core/src/main/java/org/apache/distributedlog/util/DLUtils.java (hunk 1 above)
Comment (sijie) [inline distributedlog-core/src/main/java/org/apache/distributedlog/util/DLUtils.java]: changed.

File: distributedlog-core/src/main/java/org/apache/distributedlog/util/DLUtils.java (hunk 2 above)
Comment (sijie) [inline distributedlog-core/src/main/java/org/apache/distributedlog/util/DLUtils.java]: I believe it is covered in the last "else if" branch.

File: distributedlog-core/src/test/java/org/apache/distributedlog/TestBKDistributedLogNamespace.java (hunk 3 above)
Comment (sijie) [inline distributedlog-core/src/test/java/org/apache/distributedlog/TestBKDistributedLogNamespace.java]: done

File: distributedlog-core/src/main/java/org/apache/distributedlog/util/Utils.java [hunk 4]
```diff
@@ -635,4 +636,16 @@ public static String getParent(final String path) {
         return path.substring(0, lastIndex);
     }
 
+    /**
+     * Validate the provided {@code logName}.
+     *
+     * @param logName name of the log
+     * @throws InvalidStreamNameException if the name of the log is not valid.
+     */
+    public static void validateLogName(String logName) throws InvalidStreamNameException {
```
Comment (leighst) [inline distributedlog-core/src/main/java/org/apache/distributedlog/util/Utils.java]: whats going on here?

Comment (sijie) [inline distributedlog-core/src/main/java/org/apache/distributedlog/util/Utils.java]: oh i see. this seems to be redundant. going to remove it.

**Pull request closed without merging**